doc/source/reference/c-api/array.rst

```diff
@@ -3521,6 +3551,40 @@ member of ``PyArrayDTypeMeta_Spec`` struct.
    force newly created arrays to have a newly created descriptor
    instance, no matter what input descriptor is provided by a user.
 
+.. c:macro:: NPY_DT_get_sort_function
+
+.. c:type:: int *(PyArrayDTypeMeta_GetSortFunction)(PyArray_Descr *, \
+        npy_intp sort_kind, PyArray_SortFuncWithContext **out_sort, \
+        NpyAuxData **out_auxdata, NPY_ARRAYMETHOD_FLAGS *out_flags)
+
+    .. versionadded:: 2.4
+
+    If defined, sets a custom sorting function for the DType for each of
+    the sort kinds numpy implements. Currently, sorts are always descending
+    and always use nulls to the end, and this must be checked in the
+    implementation. Returns 0 on success.
```

Comment:
Where are the sorting functions set? That is, does `out_sort` point anywhere in particular?

Reply:
The actual functions are set in the DType slots. As for `out_sort`, that is retrieved in internal NumPy code (`PyArray_(Arg)Sort`). Sorry, not sure if that's what you meant!